src/gate/parserRIL/main.cpp

```diff
@@ -0,0 +1,532 @@
+
+#include <kernel/yosys.h>
+
+namespace RTlil = Yosys::RTLIL;
+namespace YLib = Yosys::hashlib;
+
+void printModules(const RTlil::IdString module, std::ostream &out, std::ofstream &fout) {
+
+  out << module.str() << " - module of index: " << module.index_ << "\n";
+
+}
+
+void printWires(const YLib::dict<RTlil::IdString, RTlil::Wire *> &ywires,
+                std::ostream &out, std::ofstream &fout,
+                std::map<int, std::string> &inputs,
+                std::map<int, std::string> &outputs,
+                std::map<int, std::string> &wires) {
+
+  out << "Wires:" << "\n";
+  for (auto it1 = ywires.begin(); it1 != ywires.end(); ++it1) {
+    std::string temp;
+    std::string width = "u:";
+    unsigned index;
+    auto portOutput = it1->second->port_output;
+    auto portInput = it1->second->port_input;
+    index = it1->first.index_;
+    temp = it1->first.str();
+
+    out << "  " << temp << " - wire of index: " << index << "\n";
+
+    width.append(std::to_string(it1->second->width));
+    if (it1->second->width > 1) {
+      out << "    type: bus with width " << it1->second->width << "\n";
+      fout << "output " << width << " " << temp << ";\n";
+      outputs.emplace(index, temp);
+    } else {
+      out << "    type: wire" << "\n";
+    }
+    out << "    start_offset: " << it1->second->start_offset << "\n";
+    out << "    port_id: " << it1->second->port_id << "\n";
+    out << "    port_input: " << portInput << "\n";
+    if (portInput == 1) {
+      temp.erase(0, 1);
+      fout << "input " << width << " " << temp << ";\n";
+      inputs.emplace(index, temp);
+    }
+    out << "    port_output: " << portOutput << "\n";
+    if (portOutput == 1) {
+      temp.erase(0, 1);
+      fout << "output " << width << " " << temp << ";\n";
+      outputs.emplace(index, temp);
+    }
+    if (portOutput == 0 && portInput == 0) {
+      wires.emplace(index, temp);
+      fout << "wire " << width << " " << temp << ";\n";
+    }
+    out << "    upto: " << it1->second->upto << "\n\n";
+  }
+
+}
+
+std::string logicFunction(size_t type) {
+
+  std::string func;
+  if (type == ID($add).index_) {
+    func = "+";
+  }
+  if (type == ID($sub).index_) {
+    func = "-";
+  }
+  if (type == ID($and).index_) {
+    func = "&";
+  }
+  if (type == ID($or).index_) {
+    func = "|";
+  }
+  if (type == ID($xor).index_) {
+    func = "^";
+  }
+  if (type == ID($not).index_) {
+    func = "~";
+  }
+  return func;
+
+}
+
+std::string buildRIL(int root, std::map<int, std::pair<int, int>> &cell,
+                     std::map<int, std::string> &inputs,
+                     std::map<int, std::string> &typeFunc) {
+
+  bool flag1 = 0, flag2 = 0;
+  auto cellPairFirst = cell.find(root)->first;
+  auto cellPairSecond = cell.find(root)->second;
+  if (inputs.find(cellPairSecond.first) != inputs.end()) {
+    flag1 = 1;
+  }
+  if (inputs.find(cellPairSecond.second) != inputs.end()) {
+    flag2 = 1;
+  }

```

Comment:
Give proper names to the flags.
```suggestion
  auto rootCell = cell[root];
  bool flag1 = inputs.find(rootCell.first) != inputs.end();
  bool flag2 = inputs.find(rootCell.second) != inputs.end();
```